cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -86,6 +84,9 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
     }
 
     ScheduleSpecification spec = appSpec.getSchedules().get(scheduleName);
+    if (!requirementsChecker.checkSatisfied(programId, spec.getSchedule())) {
+      return Futures.<Void>immediateFuture(null);
```

Comment:
don't think we want to throw an exception here as its not really an error in cdap.  Though perhaps we would want to record somewhere why the run was skipped...